judge/management/commands/generate_report.py

```diff
@@ -0,0 +1,97 @@
+import os
+import shutil
+import sys
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from django.template.loader import get_template
+from django.core.mail import EmailMessage
+
+from judge.models import Contest, ContestParticipation
+from judge.pdf_problems import DefaultPdfMaker, PhantomJSPdfMaker, PuppeteerPDFRender, SeleniumPDFRender, \
+    SlimerJSPdfMaker
+
+
+def generate_scoreboard(contest, period):
+    problem_ids = [str(prob.id) for prob in contest.contest_problems.all().order_by('order')]
+    out = ''
+
+    for registrant in contest.registrants.all().order_by('user__user__last_name'):
+        if period and int(registrant.data['class-period']) != period:
+            continue
+
+        participation = ContestParticipation.objects.filter(contest=contest, user=registrant.user).first()
+        if not participation:
+            continue  # joined, but didn't participate
+
+        row = ''
+        for pid in problem_ids:
+            if participation.format_data and pid in participation.format_data:
+                row += '<td>%.0f</td>\n' % participation.format_data[pid]['points']
+            else:
+                row += '<td></td>\n'
+        row += '<td>%.0f</td>\n' % participation.score
+
+        out += '<tr>\n'
+        out += '<th>%s</th>\n' % registrant.user.user.get_full_name()
+        out += row
+        out += '</tr>\n'
+    return out
+
+
+class Command(BaseCommand):
+    help = 'generates a report for an ICS3U contest'
+
+    def add_arguments(self, parser):
+        parser.add_argument('key', help='key of the contest to generate a report for')
+        parser.add_argument('--period', help='ics3u class period of the contest', type=int)
+        parser.add_argument('--dry-run', help='don\'t actually email the report', action='store_true')
```

Comment:
Use double quotes first before escaping. Also put help text as the last parameter (applies to other lines as well).